web-extension/test/node-rpc.spec.js

```diff
@@ -0,0 +1,68 @@
+import NodeRPC from '../backend/node-rpc.js'
+import http from 'node:http'
+
+async function wait(ms) {
+  return new Promise((resolve) => setTimeout(resolve, ms))
+}
+
+async function createHTTPServer(fn) {
+  const server = http.createServer(fn)
+
+  await server.listen()
+
+  return {
+    url: new URL(`http://localhost:${server.address().port}/`),
+    close: () => server.close()
+  }
+}
+
+async function createServer(height = 100, timeout = 0) {
+  return createHTTPServer(async (_, res) => {
+    await wait(timeout)
+    res.end(JSON.stringify({ height }))
+  })
+}
+
+async function createFaultyServer() {
+  return createHTTPServer(async (_, res) => {
+    res.statusCode = 500
+    res.end()
+  })
+}
+
+
+describe('node-rpc', () => {
+  test('findHealthyNode - mix of nodes', async () => {
```

Comment:
Could we call the test what we are checking for? I do not know if you guys have some convention or anything but I think the test should describe what it is looking for. 

E.G check healthy node selected from a mix of healthy, unhealthy and slow nodes